llap-server/src/java/org/apache/hadoop/hive/llap/daemon/impl/TaskExecutorService.java

```diff
@@ -176,6 +181,31 @@ public TaskExecutorService(int numExecutors, int waitQueueSize,
     Futures.addCallback(future, new WaitQueueWorkerCallback());
   }
 
+  /**
+   * Sets the TaskExecutorService capacity to the new values. Both the number of executors and the
+   * queue size should be smaller than that original values, so we do not mess up with the other
```

Comment:
What other settings? Maybe the comment could be more specific?

Reply:
Container memory sizes are calculated based on the executors. Adding more executors and this way executing more containers can result in memory oversubscription.

Added more specific comment